### DISCUSSION TITLE
Typescript as an alternative language besides Python

### DISCUSSION THREAD

**Discussion by buuhvprojects** [UPBGE/upbge#1862] [Ideas/Features for UPBGE 0.3+]
I would like to suggest integrating javascript or typescript to create scripts for the engine in addition to the current python.

**Comment by IzaZed** [comment]
What would be the benefit of adding these in particular?

**Reply by buuhvprojects** [reply]
The proposal to incorporate TypeScript into UPBGE appears as a strategic measure to boost game development and strengthen our community. The inclusion of TypeScript represents not only a technological evolution, but also the expansion of this language into another ecosystem, providing a series of benefits that simplify the creation process.

**Ease of Working with TypeScript/JavaScript:**
The addition of TypeScript offers a smooth transition for developers who already have knowledge of TypeScript or JavaScript from their experiences developing web, web games, software, or mobile applications. Familiarity with the language facilitates the integration of these professionals into the UPBGE ecosystem, promoting valuable synergies between different development domains.

**Strengthening the UPBGE Community:**
The inclusion of TypeScript not only simplifies development, but also attracts a broader community of developers, leveraging those who already have experience with the language. This increase in diversity of knowledge and perspectives will strengthen our community by promoting collaboration and the exchange of innovative ideas.

**Smooth Learning Curve and Technical Creativity:**
TypeScript's gentle learning curve is a crucial factor. She encourages new developers to get involved in UPBGE, taking advantage of their familiarity with the language to speed up the adaptation process. Furthermore, the language features facilitate data manipulation, allowing for more creative approaches in solving specific game development problems.

**Comment by ClarissaGazalaEvanthe** [comment]
It's better to use dart, because it can be compiled for various platforms
